src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -1043,6 +1081,15 @@ internal static void SetupCallbacks(TraceEventDispatcher source)
 
                     // fire event
                     if (stats.JITMethodStart != null) stats.JITMethodStart(process, _method);
+
+                    // check if we should apply a lifetime limit to the method cache
+                    if (source.DataLifetimeMsec > 0 && data.TimeStampRelativeMSec >= stats.JIT.NextRelativeTimeStampMsec)
+                    {
+                        // note the next time that lifetime should be applied, to avoid cleaningup too frequently
+                        stats.JIT.NextRelativeTimeStampMsec = data.TimeStampRelativeMSec + (source.DataLifetimeMsec/2.0);
```

Comment:
Nit.   The interval between updates is of course something we have to decide, and I have argued that it is not that important (because mostly people just want to avoid being unbounded).    Thus I would argue for making less frequent, so I would suggest the 'natural' default should be 1 (that is we keep events between DataLifeTIme and 2*DataLifetime.  It is easy to describe and probably fine.    We should probably add guidance about this in the doc for the DataLifeTimeMsec that this is the expectation.  

But if you really like the 2 factor, I really don't care that much.

Reply:
As an interesting anecdote, the team that has this running did a factor of 1/2 as their starting point.  Without offering a knob, this may impact their memory sla.

Reply:
I do worry about scenarios that are overly sensitive to this factor.    In my current conception the contract does not state anything here.   

Anyway as mentioned I don't mind keeping it.    Ideally however, we make it the 'natural' (and easy to explain) value and they can hit their SLA by simply reducing the DataLifeTime value some.

Reply:
Understood.  For the concrete scenario I know there is a balance of overall memory consumption - which they want low - and the desire to retain a long enough time window to get good GC analysis - needs to be longer.  Given these are separate concepts just lowering the lifetime may not satisfy the scenario.  One place we could land is to remove the latency concept.  This tradesoff cpu time with speed.  I did measure this and in general Computers add CPU cost, which is very dependent on the load on the system.  For a system that has tremendous volume of GCs the overhead is much larger, etc.  Net-net, it sounds like you are OK (though not thrilled).  If this becomes an issue and people want to tune, perhaps we lead with reduce the lifetime.  If that is not satisfactory we can explore removing the concept of latency, as you said it is now opaque to the desire and internally tunable.

Reply:
If we have a real need for introducing the latency parameter I am not opposed to doing it.  I just want to be driven there by verified constraints (we have a real world example, and just adjusting the lifetime does not cut it).